### PR TITLE
Add BadgeSelect component

### LINKED ISSUE #323: Add BadgeSelect component
screen shot 2018-01-23 at 15 18 46

### PR DESCRIPTION
**Pull request by LucianBuzzo**
Fixes #323

change-type: minor

### CODE REVIEW

File: src/components/BadgeSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import has = require('lodash/has');
+import includes = require('lodash/includes');
+import * as React from 'react';
+import { BadgeSelectProps } from 'rendition';
+import styled from 'styled-components';
+import Badge from './Badge';
+import DropDownButton from './DropDownButton';
+import Text from './Text';
+
+interface BadgeSelectState {
```
Comment (thgreasi) [inline src/components/BadgeSelect.tsx]: NIT: how about moving this right above the component?
I was getting lost toooooo easily before starting following that rule.

File: src/components/BadgeSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	padding-right: 10px;
+	padding-top: 3px;
+	padding-bottom: 3px;
+
+	&:hover {
+		background-color: #f0f3f7;
+	}
+`;
+
+class BadgeSelect extends React.Component<BadgeSelectProps, BadgeSelectState> {
```
Comment (thgreasi) [inline src/components/BadgeSelect.tsx]: Would it make sense to name like `BadgeSelector` or `BadgeDropdown`? No strong feelings on this though.

File: src/components/BadgeSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	background: none;
+	display: block;
+	margin-top: 5px;
+	margin-bottom: 5px;
+	text-align: left;
+	min-width: 182px;
+	padding-left: 10px;
+	padding-right: 10px;
+	padding-top: 3px;
+	padding-bottom: 3px;
```
Comment (thgreasi) [inline src/components/BadgeSelect.tsx]: I think that we also need to define a cursor in here.

File: src/components/BadgeSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+							)}
+						{this.state.selected != null &&
+							!includes(this.props.items, this.state.selected) &&
+							this.state.selected}
+					</Text>
+				}
+			>
+				{this.props.items.map(item => (
+					<ButtonWrapper key={item} onClick={() => this.setSelected(item)}>
+						<Badge text={item} />
```
Comment (thgreasi) [inline src/components/BadgeSelect.tsx]: NIT: would it make sense to make these block level?

File: src/components/BadgeSelect.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+					</ButtonWrapper>
+				))}
+
+				{!!this.props.extra &&
+					this.props.extra.map(item => (
+						<ButtonWrapper key={item} onClick={() => this.setSelected(item)}>
+							<Text>{item}</Text>
```
Comment (thgreasi) [inline src/components/BadgeSelect.tsx]: Would it make sense to use a white badge for these, to properly align them with the rest of the items?

Comment (LucianBuzzo) [reply src/components/BadgeSelect.tsx]: I'd much prefer that the non-badge text is aligned with the side of the badges, not the text within them.

File: src/components/BadgeSelect.tsx (hunk 3 above)
Comment (LucianBuzzo) [reply src/components/BadgeSelect.tsx]: :+1:

File: src/components/BadgeSelect.tsx (hunk 2 above)
Comment (LucianBuzzo) [reply src/components/BadgeSelect.tsx]: I considered both of these, and settled on BadgeSelect as being the most semantic

File: src/components/BadgeSelect.tsx (hunk 4 above)
Comment (LucianBuzzo) [reply src/components/BadgeSelect.tsx]: I don't understand what you mean by 'block level', can you explain please?

File: src/components/BadgeSelect.tsx (hunk 1 above)
Comment (LucianBuzzo) [reply src/components/BadgeSelect.tsx]: can do

File: src/components/BadgeSelect.tsx (hunk 4 above)
Comment (thgreasi) [reply src/components/BadgeSelect.tsx]: I meant to have 100% width.

Comment (LucianBuzzo) [reply src/components/BadgeSelect.tsx]: Ah ok, the designs are not 100% width

**Pull request merged**